alerting/src/main/kotlin/org/opensearch/alerting/model/Monitor.kt

```diff
@@ -55,7 +56,8 @@ data class Monitor(
     val schemaVersion: Int = NO_SCHEMA_VERSION,
     val inputs: List<Input>,
     val triggers: List<Trigger>,
-    val uiMetadata: Map<String, Any>
+    val uiMetadata: Map<String, Any>,
+    val dataSources: DataSources? = null
```

Comment:
the changes i suggested in `DataSources` will allow us to remove the null check here e.g. from

```
val dataSources: DataSources? = null
```
to
```
val dataSources: DataSources = DataSources()
```